### PR TITLE
feature/disableSetAccessControlAllowOrigin

### PR DESCRIPTION
**Pull request by miguelromero2000**
- Added the possibility to disable having to set the header "Access-Control-Allow-Origin" by setting the option to false - allow_origin: false
- Added test testPreflightedRequestWithoutOriginDoesNotReturnOriginHeader

### CODE REVIEW

**Comment by miguelromero2000** [comment]
This feature is needed when the header AccessControlAllowOrigin is handled and set in the web server config, in order to avoid 2 headers to be set.

**Comment by miguelromero2000** [comment]
By the way, I do not really know why the hhvm build is failing, I think there is something weird in that setup, and nothing related to my PR:
1) Nelmio\Tests\DependencyInjection\CorsConfigurationProviderPassTest::testCollectProviders
Error: unknown class PHPUnit\Framework\Constraint\Constraint
/home/travis/build/nelmio/NelmioCorsBundle/vendor/matthiasnoback/symfony-dependency-injection-test/PhpUnit/DefinitionHasArgumentConstraint.php:10
/home/travis/build/nelmio/NelmioCorsBundle/vendor/matthiasnoback/symfony-dependency-injection-test/PhpUnit/AbstractContainerBuilderTestCase.php:166
/home/travis/build/nelmio/NelmioCorsBundle/Tests/DependencyInjection/CorsConfigurationProviderPassTest.php:56
ERRORS!
Tests: 16, Assertions: 42, Errors: 1.

File: EventListener/CorsListener.php [hunk 1]
```diff
@@ -136,6 +136,10 @@ protected function getPreflightResponse(Request $request, array $options)
             $response->headers->set('Access-Control-Max-Age', $options['max_age']);
         }
 
+        if (0 === count($options['allow_origin'])) {
```
Comment (nicolas-grekas) [inline EventListener/CorsListener.php]: missing is_array() check before

**Comment by Seldaek** [comment]
This needs a rebase on latest master and the test should be updated to use the new event classes etc to bring it in line then it can be merged for 2.0

**Comment by Seldaek** [comment]
Just a question though, what's the point of this, versus simply not using the bundle? Is it to allow prod vs dev config to be different?

**Pull request closed without merging**